### ISSUE TITLE
[Markdown Parser - Python] Inline styling is wrong

### ISSUE THREAD

**Issue by Mariosyian** [Mariosyian/benchmarking-programming-languages#11]
When adding inline styling (i.e. bold, italic, etc) the parsing breaks and wraps the wrong text in the respective tag.

This is seen in places where the start of the line is plain text, but not when the start of the line is the inline characters themselves, or another character such as a '#' (header tag).

e.g.
```
**This is bold.**
# This is an h1 with **bold text**

These will parse correctly to
<b>This is bold.</b>
<h1> This is an h1 with <b>bold text</b></h1>
```
but these will cause errors
```
This is plain text with **bold text** in the middle.
This is plain text with **bold text** and an image ![foo-bar](www.foo.bar).
```

**Comment by Mariosyian** [comment]
Introduced in 0204044144d88ef9b74c32bb21aeb7f0f9f3cfaa

**Comment by Mariosyian** [comment]
Issue closed by commit 8fda5b13f6406541056dc869b71f9fa60067b708
